alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/cognitiveagents/characteristics/cognitive/AbstractOrientingBehavior.kt

```diff
@@ -365,6 +401,6 @@ abstract class AbstractOrientingBehavior<T, P, A, N, E, M, F>(
             environment.graph().outgoingEdgesOf(area).distinct().count() <= 1
 
     companion object {
-        private const val destinationWeight = 0.1
+        private const val finalDestinationWeight = 0.1
```

Comment:
Why 0.1? Looks like this is an hidden parameter...

Reply:
I removed it, so we have fewer parameters.

Reply:
So this is no longer a parameter? Was it unused? Where have this 0.1 gone?

Reply:
That parameter came from Andresen's paper and it was used to lower the weight of edges leading to final destinations. In brief: when weighting edge, the final destination factor assumed value `0.1` if the edge leaded to a final destination, `1.0` otherwise (since factors are multiplied, `1.0` is the identity-value or something - can't remember how it's called). This allowed pedestrians to consider (and take) exits they eventually found along the way. The idea behind `0.1` (at least this is what I suppose, as Andresen didn't documented that so much) is that it should lower the weight of an exit so as to force a pedestrian to choose that edge, but without voiding the weight (i.e. it couldn't simply be `0`), so as to allow different weights based on other factors in case of multiple exits.
The final destination factor was a good way of taking into account exits found along the way. However, I removed it. Now our pedestrians explicitly look for exits they may found along the way (in particular, whenever they're in a new room). If one or more exits are found, they are weighted with a _different_ weighting function, and one of them is taken. The weighting function for exits at present takes into account the distance of an exit and the congestion. This allows us to consider different factors when weighting an exit, which is good because e.g. we don't care about the cognitive map factor, which was previously taken into account as well.
What do you think?

Reply:
If the parameter is gone, then it's ok. From the diff it looks to me it's just renamed, though.
If this parameter still circulates (but this is general for every parameter), then:
* reify it as public in a companion object
* add it as parameter in the @jvmOverloads constructor, defaulting to the default value of the companion
* document the default value citing the source of the default value, so that the value origin can be traced, and does not look like something magic that we threw in to make the machinery work.

Reply:
Ok, in the future I'll do that.
Concerning our `0.1`, it's not renamed but completely removed. If you want, point me the code that made you think so.